tsdb/db_test.go

```diff
@@ -5293,6 +5293,174 @@ func test_ChunkQuerier_OOOQuery(t *testing.T,
 	}
 }
 
+func TestOOONativeHistogramsWithCounterResets(t *testing.T) {
+	for name, scenario := range sampleTypeScenarios {
+		t.Run(name, func(t *testing.T) {
+			if name == "integer histogram" || name == "float histogram" {
+				testOOONativeHistogramsWithCounterResets(t, scenario)
+			}
+		})
+	}
+}
+
+func testOOONativeHistogramsWithCounterResets(t *testing.T, scenario sampleTypeScenario) {
+	opts := DefaultOptions()
+	opts.OutOfOrderCapMax = 30
+	opts.OutOfOrderTimeWindow = 24 * time.Hour.Milliseconds()
+	opts.AllowOverlappingCompaction = false
+
+	type filterFunc func(v int64) bool
+	defaultFilterFunc := func(v int64) bool { return false }
+
+	lbls := labels.FromStrings("foo", "bar1")
+	minutes := func(m int64) int64 { return m * time.Minute.Milliseconds() }
+
+	type sampleBatch struct {
+		from                 int64
+		until                int64
+		filter               filterFunc
+		expCounterResetHints []histogram.CounterResetHint
+	}
+
+	tests := []struct {
+		name            string
+		queryMin        int64
+		queryMax        int64
+		appendFunc      func(appender storage.Appender, lbls labels.Labels, ts, value int64) error
+		batches         []sampleBatch
+		expectedSamples []chunks.Sample
+	}{
+		{
+			name:     "Counter reset right within in-order samples",
```

Comment:
```suggestion
			name:     "Counter reset within in-order samples",
```

Not sure if "right" should be in the test name?